packages/components/src/FilterDatePicker/_docs/FilterDatePicker.stories.tsx

```diff
@@ -0,0 +1,332 @@
+import React, { useEffect, useState } from "react"
+import { Meta, StoryFn } from "@storybook/react"
+import Highlight from "react-highlight"
+import { Paragraph } from "@kaizen/typography"
+import { renderTriggerControls } from "~components/Filter/_docs/controls/renderTriggerControls"
+import {
+  FilterButton,
+  FilterButtonProps,
+  FilterButtonRemovable,
+} from "~components/FilterButton"
+import {
+  DateValidationResponse,
+  FilterDatePicker,
+  ValidationMessage,
+} from "../index"
+import { FilterDatePickerField } from "../subcomponents/FilterDatePickerField"
+import { defaultMonthControls } from "./controls/defaultMonthControls"
+import { disabledDaysControls } from "./controls/disabledDaysControls"
+import { validationControls } from "./controls/validationControls"
+
+export default {
+  title: "Components/Filter Date Picker",
+  component: FilterDatePicker,
+  args: {
+    label: "Date",
+    locale: "en-AU",
+  },
+  argTypes: {
+    ...defaultMonthControls,
+    ...validationControls,
+    disabledDays: disabledDaysControls,
+    renderTrigger: renderTriggerControls,
+    locale: {
+      options: ["en-US", "en-AU"],
+      control: { type: "radio" },
+    },
+    isOpen: { control: "disabled" },
+    selectedDate: {
+      options: ["None", "Date"],
+      control: {
+        type: "select",
+        labels: {
+          None: "undefined",
+        },
+      },
+      mapping: {
+        None: undefined,
+        Date: new Date(),
+      },
+    },
+    description: {
+      control: "text",
+    },
+  },
+} satisfies Meta<typeof FilterDatePicker>
+
+const sampleCode = `
+// This code is not connected to the controls of the attached component.
+// @note: If you want a removable button, use the commented out code instead.
+
+import {
+  FilterButton,
+  FilterButtonProps,
+//  FilterButtonRemovable,
+  FilterDatePicker,
+} from "@kaizen/components"
+
+const [isOpen, setIsOpen] = useState<boolean>(false)
+const [date, setDate] = useState<Date | undefined>()
+
+return (
+  <FilterDatePicker
+    id="filter-dp--default"
+    label="Dates"
+    locale="en-AU"
+    renderTrigger={(triggerButtonProps: FilterButtonProps): JSX.Element => (
+      <FilterButton {...triggerButtonProps} />
+    )}
+    // renderTrigger={(triggerButtonProps: FilterButtonProps): JSX.Element => (
+    //   <FilterButtonRemovable
+    //     triggerButtonProps={{ ...triggerButtonProps }}
+    //     removeButtonProps={{
+    //       onClick: (): void => undefined,
+    //     }}
+    //   />
+    // )}
+    isOpen={isOpen}
+    setIsOpen={setIsOpen}
+    selectedDate={range}
+    onDateChange={setRange}
```

Comment:
```suggestion
    selectedDate={date}
    onDateChange={setDate}
```

You missed this part of my previous comment 😂